### PR TITLE
fix an issue with prettify query

### PR DESCRIPTION
**Pull request by dmitryk-dk**
Fixed issue with prettify query if if the query includes Grafana variables in the lookbehind window.

See this issue: https://github.com/VictoriaMetrics/grafana-datasource/issues/166

### CODE REVIEW

File: src/components/PrettifyQuery.tsx [hunk 1]
```diff
@@ -26,10 +37,21 @@ const PrettifyQuery: FC<Props> = ({
   const handleClickPrettify = async () => {
     setLoading(true)
     try {
-      const response = await datasource.prettifyRequest(query.expr)
+      let { expr } = query;
```
Comment (hagen1778) [inline src/components/PrettifyQuery.tsx:40]: Could you please add tests for this?
Just want to verify that conversion from original query to modified and back is correct.

**Review by hagen1778 — CHANGES_REQUESTED** [review]

File: src/components/PrettifyQuery.tsx (hunk 1 above)
Comment (dmitryk-dk) [reply src/components/PrettifyQuery.tsx:41]: The logic was improved, and tests were added. 
In the test, I used a mock response. It is not prettified, just normal functions. 
But all tests shows that grafana variables correctly removed from the the request with default lookbehind window and after it correctly replaced with grafana variable

Comment (Loori-R) [reply src/components/PrettifyQuery.tsx:41]: I'm not sure if such requests can be used, but they can be executed without errors.
The logic works incorrectly:
1) If there are multiple variables in the expression,
2) If the expression includes `[1i]`.

````javascript
   rate(metric_name{mode="idle"}[$__interval]) + up{instance="[1i]"} 
// rate(metric_name{mode="idle"}[$__interval]) + up{instance="[$__interval]"}

   rate(metric_name[$__interval]) + rate(metric_name[$__range]) 
// rate(metric_name[$__range]) + rate(metric_name[$__range])

````

Comment (dmitryk-dk) [reply src/components/PrettifyQuery.tsx:41]: i think it is a valid points, will update tests and logic

**Review by Loori-R — APPROVED** [review]
LGTM!

**Review by hagen1778 — APPROVED** [review]
Thanks!

**Comment by hagen1778** [comment]
@dmitryk-dk @Loori-R before merging this, do you think we should have a common approach for such issues like this and https://github.com/VictoriaMetrics/grafana-datasource/issues/167?

Or https://github.com/VictoriaMetrics/grafana-datasource/issues/167 should be solved differently?

**Comment by dmitryk-dk** [comment]
> @dmitryk-dk @Loori-R before merging this, do you think we should have a common approach for such issues like this and #167?
> 
> Or #167 should be solved differently?

I think we can move my part of the logic to the 

> @dmitryk-dk @Loori-R before merging this, do you think we should have a common approach for such issues like this and #167?
> 
> Or #167 should be solved differently?

I think we can merge both of the PRs, but in the follow-up commit, we can check if we can use only one function to handle different variables in one function like `this.enhanceExprWithAdHocFilters(expr)` in the PR you mentioned.

**Pull request merged**